src/main/java/jenkins/plugins/openstack/compute/SlaveOptionsDescriptor.java

```diff
@@ -483,4 +484,18 @@ public FormValidation doCheckFsRoot(
     private @Nonnull String def(@CheckForNull Object val) {
         return val == null ? "" : ("Inherited value: " + val);
     }
+
+    /**
+     * Returns the list of {@link NodePropertyDescriptor} appropriate for the
+     * {@link JCloudsSlave}s that are created from these options.
+     *
+     * @return the filtered list
+     */
+    @Nonnull
+    @Restricted(NoExternalUse.class) // used by Jelly EL only
+    public List<NodePropertyDescriptor> getNodePropertiesDescriptors() {
+        final Jenkins j = Jenkins.get();
+        final JCloudsSlaveDescriptor jcsd = (JCloudsSlaveDescriptor) j.getDescriptor(JCloudsSlave.class);
+        return jcsd.nodePropertyDescriptors(null);
```

Comment:
When I do a local build, I see that findbugs is showing
```
[INFO] Possible null pointer dereference in jenkins.plugins.openstack.compute.SlaveOptionsDescriptor.getNodePropertiesDescriptors() due to return value of called method [jenkins.plugins.openstack.compute.SlaveOptionsDescriptor, jenkins.plugins.openstack.compute.SlaveOptionsDescriptor] Dereferenced at SlaveOptionsDescriptor.java:[line 499]Known null at SlaveOptionsDescriptor.java:[line 498] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
```

I've checked the code and it's correct - `getDescriptor(...)` is annotated with `@CheckForNull`  - _but_ it should never return null in this case.
If it does return null when given `JCloudsSlave.class` as an argument, our plugin (and probably Jenkins as a whole) is likely to be broken for other reasons, so it probably doesn't matter if this method throws a NPE at that point :wink: 
I've googled and found that I could add code saying
```
    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings(
            value="NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE", 
            justification="It's ok to NPE if *our* slave descriptor is missing")
```
in front of the method and that'll then prevent this findbugs complaint from appearing.
WDYT?

Reply:
Better use `#getDescriptorOrDie()` that would throw an exception.

Reply:
Fixed in https://github.com/jenkinsci/openstack-cloud-plugin/pull/270/commits/73df874c9fbedb780b913328ca0f577dfeeb73e4